metadata/src/utils/validation.rs

```diff
@@ -104,6 +104,30 @@ fn get_variant_hash(
     concat_and_hash2(&variant_name_bytes, &variant_field_bytes)
 }
 
+fn get_type_def_variant_hash(
+    registry: &PortableRegistry,
+    variant: &TypeDefVariant<PortableForm>,
+    specific_pallets: &Option<Vec<&str>>,
```

Comment:
But, a follow-up; why is this actually needed at all? Since we expect to call `retain_pallets_in_runtime_outer_types`, which strips out unwanted variants already, then do we need to have any "specific pallets" type checking in `get_extrinsic_hash`?

Reply:
Yep, that makes sense!

Users may still want to compare some partially stripped metadata to the full metadata of a node. 
For those cases, the stripped metadata will contain only some specific variants for the `RuntimeCall`, while the full metadata will differ containing all pallets as variants.

The metadata also includes the `RuntimeError` and `RuntimeEvent` enums that are similar to the `RuntimeCall`.
The first 2 enums aren't hashed specifically in the validation phase, since the values of those variants are hashed indirectly while passing through the pallet's metadata.

The main reason for hashing the `RuntimeCall` explicitly here is because it is a component of the extrinsic's metadata. We could potentially generate the same extrinsic hash for different extrinsic metadatas (depending on the Runtime variants).

I believe we have two options for this:
- don't hash the `RuntimeCall` as part of the extrinsic metadata and rely on the pallets's parts to hash indirectly the variant's value; and have a slightly inconsistent extrinsic metadata hash
- hash the `RuntimeCall`; and have some extra steps to filter the variants

I'd love to get your thoughts on this! I'm not super sure if we may need to maintain hashing compatibility at the `ExtrinsicMetadata`'s level in the future

Reply:
Summary because that was longer than planned:

I think we should hash (with `only_these_pallets` if given) the `outer_enum` types, and if we do that I don't think we need to hash the `extrinsic_metadata.call_ty` one, since it's the same I think?

Reply:
Yep, that makes a lot of sense, sounds like a plan!